src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/VersionCompatibility/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -185,17 +185,29 @@ protected virtual async Task<bool> IsPreviewSDKInUseAsync()
             return settings.GetValueOrDefault<bool>(UsePreviewSdkSettingKey);
         }
 
-        private async Task<ISetupInstanceCatalog> TryGetSetupInstanceAsync()
+        private Task<bool> IsPrereleaseAsync()
         {
-            await _threadHandling.Value.SwitchToUIThread();
             var vsSetupConfig = new SetupConfiguration();
-            ISetupInstance setupInstance = vsSetupConfig.GetInstanceForCurrentProcess();
-            if (setupInstance is ISetupInstanceCatalog setupInstanceCatalog)
+            if (vsSetupConfig.GetInstanceForCurrentProcess() is ISetupInstanceCatalog setupInstanceCatalog)
             {
-                return setupInstanceCatalog;
+                return TryQueryCOMObject(setupInstanceCatalog);
             }
 
-            return null;
+            return TaskResult.False;
+        }
+
+        private async Task<bool> TryQueryCOMObject(ISetupInstanceCatalog setupInstanceCatalog)
+        {
+            try
+            {
+                await _threadHandling.Value.SwitchToUIThread();
+                return setupInstanceCatalog.IsPrerelease();
+            }
+            catch (Exception)
+            {
+                // COM object may not be queryable yet.
```

Comment:
What isn't the COM object queryable? If its early on in the VS process, don't this mean this check won't work for the first project someone opens, ever, but will work for subsequent loads?

Sorry if those are silly questions, but I don't fully understand why this COM error is occurring.

Reply:
Yeah, I will admit to not fully understanding some of what this cause the COm query to fail in the startup. If this happens it is possible for the project file to not notify immediately